src/main/kotlin/org/rust/ide/hints/RsInlayParameterHintsProvider.kt

```diff
@@ -112,6 +116,29 @@ enum class HintType(desc: String, enabled: Boolean) {
         }
 
         override fun isApplicable(elem: PsiElement): Boolean = elem is RsForExpr
+    },
+
+    /* In case of fields without any expressions
+       PSI is destructed in this place, so only hints before error element will be shown */
+    STRUCT_LITERAL_TYPE_HINT("Show type hints for fields in struct literal", true) {
+        override fun provideHints(elem: PsiElement): List<InlayInfo> {
+            val element = elem as? RsStructLiteral ?: return emptyList()
+            val declaration = element.path.reference.resolve() as? RsFieldsOwner ?: return emptyList()
+            val subst = element.type.typeParameterValues
+            val hints = element.structLiteralBody.structLiteralFieldList.mapNotNull { field ->
+                (declaration.namedFields
+                    .find { it.name.equals(field.referenceName) }
+                    ?.typeReference?.type?.substitute(subst)
+                    ?.shortPresentableText
+                    ?: return@mapNotNull null) to (
```

Comment:
it is very unobvious construction. Let's extract left and right parts of `to` expression to variables.